server/store/peer_store.go

```diff
@@ -15,15 +15,39 @@ var peerStoreDrivers = make(map[string]PeerStoreDriver)
 
 // PeerStore represents an interface for manipulating peers.
 type PeerStore interface {
+	// PutSeeder adds a seeder for the infoHash to the PeerStore
 	PutSeeder(infoHash chihaya.InfoHash, p chihaya.Peer) error
+	// DeleteSeeder removes a seeder for the infoHash from the PeerStore
 	DeleteSeeder(infoHash chihaya.InfoHash, p chihaya.Peer) error
 
+	// PutLeecher adds a leecher for the infoHash to the PeerStore
 	PutLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error
+	// DeleteLeecher removes a leecher for the infoHash from the PeerStore
 	DeleteLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error
 
+	// GraduateLeecher promotes a peer from a leecher to a seeder for the infoHash within the PeerStore
 	GraduateLeecher(infoHash chihaya.InfoHash, p chihaya.Peer) error
+	// AnnouncePeers returns a list of both ipv4, and ipv6 peers for an announce
+	//
+	// If seeder is true then the peers returned will only be leechers, the ammount
+	// of leechers returned will be the smaller value of numWant or the available
+	// leechers.
+	// If it is false then seeders will be returned up until numWant or the available
+	// seeders, whichever is smaller. If the available seeders is less than numWant
+	// then peers are returned until numWant or they run out.
 	AnnouncePeers(infoHash chihaya.InfoHash, seeder bool, numWant int) (peers, peers6 []chihaya.Peer, err error)
+	// CollectGarbage deletes peers from the peerStore which are older than the cutoff time
 	CollectGarbage(cutoff time.Time) error
+
+	// GetSeeders gets all the seeders for a particular infoHash
+	GetSeeders(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
+	// GetLeechers gets all the leechers for a particular infoHash
+	GetLeechers(infoHash chihaya.InfoHash) (peers, peers6 []chihaya.Peer, err error)
```

Comment:
The only reason we have `GetSeeders` and `GetLeechers` is for the API, right? `AnnouncePeers` should handle it for the tracker aspect. I wonder if we could instead be clever and make it so that `AnnouncePeers` with `-1` for numWant returns all. Then we could just call it twice with seeder as `true` and `false` and then de-duplicate what we have. It makes the API slightly less efficient, but lessens the burden on implementing drivers. The API shouldn't be used nearly as much as the Tracker itself either.